python/pyiceberg/io/pyarrow.py

```diff
@@ -515,6 +529,14 @@ def _file_to_table(
         if pyarrow_filter is not None:
             arrow_table = arrow_table.filter(pyarrow_filter)
 
+        if len(positional_deletes) > 0:
+            # When there are positional deletes, create a filter mask
+            mask = [True] * len(arrow_table)
+            for buffer in positional_deletes:
+                for pos in buffer:
+                    mask[pos.as_py()] = False
+            arrow_table = arrow_table.filter(mask)
```

Comment:
It seems wasteful to create a Python list for this, but I'm having trouble finding a better solution. It looks like we can create an array of nulls and replace null with True, like this:

```python
null_arr = pa.nulls(10, type=pa.bool_())
true_arr = pa.fill_null(True)
```

Unfortunately, I don't see an easy way to set individual values to False in the resulting array.

Reply:
What do you think of the following:
```
def generator():
    itr = iter(positional_deletes)
    next_delete = next(itr)
    for pos in range(len(arrow_table)):
        if pos == next_delete:
            yield True
            next_delete = next(itr)
        else:
            yield False
mask = pa.array(generator(), type=pa.bool_())
arrow_table = arrow_table.filter(mask)
```
This creates an iterator (`range` in Python is also an iterator) that will materialize the array and this way we avoid creating large collections on the Python side.